src/Chapter4.hs

```diff
@@ -628,8 +664,11 @@ Can you implement a monad version of AND, polymorphic over any monad?
 
 🕯 HINT: Use "(>>=)", "pure" and anonymous function
 -}
+
+
 andM :: (Monad m) => m Bool -> m Bool -> m Bool
-andM = error "andM: Not implemented!"
+andM ma mb = ma >>= (\a -> fmap ((&&) a) mb)
+-- andM ma mb = (fmap (\a -> (&&) a) ma) <*> mb
```

Comment:
The ordinary `&&` if lazy on the second argument. Meaning that if it meets `False` as the first argument, there is no need to calculate another arg. In `andM` we would like to keep the same behaviour, so the

```
Just False `andM` _
```

should be `Just False` :slightly_smiling_face: 

If you don't use the `>>=` operator, then the `Monad` constraint is too strong, and it is enough to specify only `Applicative`. However, with `Applicative` you are not able to implement a lazier version of `andM` due to the nature of the `Applicative`. `Monad` though, can stop when needed, so you can use that for laziness 🙂

Reply:
Okay, a little hard to understand. But, I will try to make the lazier version of this and I guess the `pure` would be helpful with this.

Reply:
Done